### PR TITLE
Add map and reduce to the cheatsheet

### PR DESCRIPTION
**Pull request by holguinj**
These functions are super important but kind of confusing, so this adds a couple of fairly simple examples. Closes Issue 149.

I also fixed what I'm pretty sure was a typo in the function section.

### CODE REVIEW

File: outline/cheatsheet.md [hunk 1]
```diff
@@ -88,12 +88,34 @@
...
 
+## Map and Reduce
+
+```clj
+; apply a function to every element in a collection
+(map inc [1 2 3 4])
+;=> (2 3 4 5)
+; Similar to [(inc 1) (inc 2) (inc 3) (inc 4)]
+```
+**`map`** takes a function and a collection, then applies the function to each element in the collection in order. It returns the new collection.
```
Comment (davidchambers) [inline outline/cheatsheet.md]: I'd drop "in order". When dealing with pure functions, one shouldn't be concerned with the order in which the function is applied to the elements.

File: outline/cheatsheet.md [hunk 2]
```diff
@@ -88,12 +88,34 @@
...
+
+```clj
+; reduce a collection to one value using a function
+(reduce + [1 3 5 7])
+;=> 16
+; Similar to (+ 1 3) ;=> 4
+;            (+ 4 5) ;=> 9
+;            (+ 9 7) ;=> 16
```
Comment (davidchambers) [inline outline/cheatsheet.md]: This is nice!

File: outline/cheatsheet.md [hunk 3]
```diff
@@ -88,12 +88,34 @@
...
+```
+
+**`reduce`** takes a function and a collection. First, it applies the function to the first element and the second element, then it applies the function to the *result* of that, plus the third element. It continues rolling up the results until it gets to the last thing in the collection.
```
Comment (davidchambers) [inline outline/cheatsheet.md]: I suggest _… until all the elements in the collection have been “reduced”._

**Comment by kpfell** [comment]
An added note: we should update [the PDF](https://github.com/ClojureBridge/curriculum/blob/master/ClojureBridgeCheatsheet-v1.pdf) along with this PR! :tada:

**Comment by yokolet** [comment]
Since all contents were moved to gh-pages, I merged this pull request manually.

BTW, how to create PDF from cheatsheet.md ?

**Pull request merged**